src/openvr/__init__.py

```diff
@@ -1,6947 +1,6949 @@
-VRTrackedCameraError_InvalidSharedTextureHandle = ENUM_VALUE_TYPE(110)
-VRTrackedCameraError_FailedToGetGLTextureId = ENUM_VALUE_TYPE(111)
-VRTrackedCameraError_SharedTextureFailure = ENUM_VALUE_TYPE(112)
-VRTrackedCameraError_NoFrameAvailable = ENUM_VALUE_TYPE(113)
-VRTrackedCameraError_InvalidArgument = ENUM_VALUE_TYPE(114)
-VRTrackedCameraError_InvalidFrameBufferSize = ENUM_VALUE_TYPE(115)
-
-EVRTrackedCameraFrameLayout = ENUM_TYPE
-EVRTrackedCameraFrameLayout_Mono = ENUM_VALUE_TYPE(1)
-EVRTrackedCameraFrameLayout_Stereo = ENUM_VALUE_TYPE(2)
-EVRTrackedCameraFrameLayout_VerticalLayout = ENUM_VALUE_TYPE(16)
-EVRTrackedCameraFrameLayout_HorizontalLayout = ENUM_VALUE_TYPE(32)
-
-EVRTrackedCameraFrameType = ENUM_TYPE
-VRTrackedCameraFrameType_Distorted = ENUM_VALUE_TYPE(0)
-VRTrackedCameraFrameType_Undistorted = ENUM_VALUE_TYPE(1)
-VRTrackedCameraFrameType_MaximumUndistorted = ENUM_VALUE_TYPE(2)
-MAX_CAMERA_FRAME_TYPES = ENUM_VALUE_TYPE(3)
-
-EVRDistortionFunctionType = ENUM_TYPE
-VRDistortionFunctionType_None = ENUM_VALUE_TYPE(0)
-VRDistortionFunctionType_FTheta = ENUM_VALUE_TYPE(1)
-VRDistortionFunctionType_Extended_FTheta = ENUM_VALUE_TYPE(2)
-MAX_DISTORTION_FUNCTION_TYPES = ENUM_VALUE_TYPE(3)
-
-EVSync = ENUM_TYPE
-VSync_None = ENUM_VALUE_TYPE(0)
-VSync_WaitRender = ENUM_VALUE_TYPE(1)
-VSync_NoWaitRender = ENUM_VALUE_TYPE(2)
-
-EVRMuraCorrectionMode = ENUM_TYPE
-EVRMuraCorrectionMode_Default = ENUM_VALUE_TYPE(0)
-EVRMuraCorrectionMode_NoCorrection = ENUM_VALUE_TYPE(1)
-
-Imu_OffScaleFlags = ENUM_TYPE
-OffScale_AccelX = ENUM_VALUE_TYPE(1)
-OffScale_AccelY = ENUM_VALUE_TYPE(2)
-OffScale_AccelZ = ENUM_VALUE_TYPE(4)
-OffScale_GyroX = ENUM_VALUE_TYPE(8)
-OffScale_GyroY = ENUM_VALUE_TYPE(16)
-OffScale_GyroZ = ENUM_VALUE_TYPE(32)
-
-EVRApplicationError = ENUM_TYPE
-VRApplicationError_None = ENUM_VALUE_TYPE(0)
-VRApplicationError_AppKeyAlreadyExists = ENUM_VALUE_TYPE(100)
-VRApplicationError_NoManifest = ENUM_VALUE_TYPE(101)
-VRApplicationError_NoApplication = ENUM_VALUE_TYPE(102)
-VRApplicationError_InvalidIndex = ENUM_VALUE_TYPE(103)
-VRApplicationError_UnknownApplication = ENUM_VALUE_TYPE(104)
-VRApplicationError_IPCFailed = ENUM_VALUE_TYPE(105)
-VRApplicationError_ApplicationAlreadyRunning = ENUM_VALUE_TYPE(106)
-VRApplicationError_InvalidManifest = ENUM_VALUE_TYPE(107)
-VRApplicationError_InvalidApplication = ENUM_VALUE_TYPE(108)
-VRApplicationError_LaunchFailed = ENUM_VALUE_TYPE(109)
-VRApplicationError_ApplicationAlreadyStarting = ENUM_VALUE_TYPE(110)
-VRApplicationError_LaunchInProgress = ENUM_VALUE_TYPE(111)
-VRApplicationError_OldApplicationQuitting = ENUM_VALUE_TYPE(112)
-VRApplicationError_TransitionAborted = ENUM_VALUE_TYPE(113)
-VRApplicationError_IsTemplate = ENUM_VALUE_TYPE(114)
-VRApplicationError_SteamVRIsExiting = ENUM_VALUE_TYPE(115)
-VRApplicationError_BufferTooSmall = ENUM_VALUE_TYPE(200)
-VRApplicationError_PropertyNotSet = ENUM_VALUE_TYPE(201)
-VRApplicationError_UnknownProperty = ENUM_VALUE_TY
```

Comment:
The original API function has the following footprint:
```c++
virtual bool GetDefaultApplicationForMimeType( const char *pchMimeType, VR_OUT_STRING() char *pchAppKeyBuffer, uint32_t unAppKeyBufferLen ) = 0;
```
I.e. it returns `bool` and there seems to be no way to figure out the right buffer size, or what the return value means, unless one tries it out. There is also no `error` value in the function declaration either. So this type of function is not correctly translated (there are more of them).

Reply:
We should spin these cases out into a separate issue. Hopefully we can find or construct some examples of these functions being used correctly from C++, so we can reason properly about how they should be used from python. It could also be helpful to see what the C# bindings do for these functions. 

Most of these functions have the `VR_OUT_STRING` clang annotation, which is great, if it is used consistently, because the translation module uses clanglib and these annotations are available at our fingertips.

Reply:
To be honest, I have no idea, how these functions are supposed to be used correctly. The header itself does not give much of the clue and I do not have any experience with them either. The only thing I can currently think of is asking at OpenVR project, or do some wild guessing based on anecdotal experience. How would like to handle it?